hazelcast/src/main/java/com/hazelcast/concurrent/atomiclong/AtomicLongContainer.java

```diff
@@ -17,15 +17,30 @@
 package com.hazelcast.concurrent.atomiclong;
 
 import com.hazelcast.config.AtomicLongConfig;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.SplitBrainAwareDataContainer;
+import com.hazelcast.spi.SplitBrainMergeEntryView;
+import com.hazelcast.spi.SplitBrainMergePolicy;
 
-public class AtomicLongContainer {
+import static com.hazelcast.spi.merge.SplitBrainEntryViews.createSplitBrainMergeEntryView;
 
+public class AtomicLongContainer implements SplitBrainAwareDataContainer<Void, Long, Long> {
+
+    private final NodeEngine nodeEngine;
+    private final String name;
     private final AtomicLongConfig config;
 
+    private boolean isModified;
```

Comment:
This field may be lost if there is a migration in the bigger cluster before the merge (e.g. member added).

Reply:
I just applied a similar approach like we I for the Lock prototype and got rid of the whole field. I'm looking in the operation before calling `getLongContainer()` if it exists and put that information in the unused `key` field of the `SplitBrainMergeEntryView` (not super elegant, but it shouldn't hurt anyone).

```
    @Override
    public void run() throws Exception {
        AtomicLongService service = getService();
        boolean isExistingContainer = service.containsAtomicLong(name);

        SplitBrainMergeEntryView<Boolean, Long> mergingEntry
            = createSplitBrainMergeEntryView(isExistingContainer, mergingValue);
        backupValue = getLongContainer().merge(mergingEntry, mergePolicy);
    }
```